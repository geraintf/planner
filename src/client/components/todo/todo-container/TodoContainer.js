import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import AddTodo from '../add-todo/AddTodo';
import TodoList from '../todo-list/TodoList';
import SortableTodoList from '../SortableTodoList';

const TodoContainer = ({ addTodo, sortable, toggleTodo, items, moveTodo }) => {

  const list = sortable
    ? (
      <SortableTodoList
        toggleTodo={toggleTodo}
        items={items}
        moveTodo={moveTodo}
      />)
    : (<TodoList toggleTodo={toggleTodo} items={items} />);

  return (
    <Fragment>
      { list }
      <AddTodo addTodo={addTodo} />
    </Fragment>
  );

};

TodoContainer.propTypes = {
  addTodo: PropTypes.func.isRequired,
  sortable: PropTypes.bool,
  toggleTodo: PropTypes.func.isRequired,
  moveTodo: PropTypes.func.isRequired,
  items: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
    text: PropTypes.string,
    completed: PropTypes.bool
  })).isRequired,
};

TodoContainer.defaultProps = {
  sortable: false
};

export default TodoContainer;
