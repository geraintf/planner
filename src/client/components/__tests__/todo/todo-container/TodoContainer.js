import React from 'react';
import  { shallow } from 'enzyme';

import TodoContainer from '../../../todo/todo-container/TodoContainer';

describe('TodoContainer', () => {
  let output;
  let props;
  beforeEach(() => {
    props = {
      addTodo: () => {},
      toggleTodo: () => {},
      moveTodo: () => {},
      items: [{
        id: '123',
        text: 'some stuff',
        completed: true
      }]
    };
    output = shallow(<TodoContainer {...props} />);
  });

  it('should render correctly', () => {
    expect(output).toMatchSnapshot();
  });
});
