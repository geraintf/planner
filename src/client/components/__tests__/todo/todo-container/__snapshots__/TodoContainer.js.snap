// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoContainer should render correctly 1`] = `
<React.Fragment>
  <SortableTodoList
    items={
      Array [
        Object {
          "completed": true,
          "id": "123",
          "text": "some stuff",
        },
      ]
    }
    moveTodo={[Function]}
    toggleTodo={[Function]}
  />
  <AddTodo
    addTodo={[Function]}
  />
</React.Fragment>
`;
